var/spack/repos/builtin/packages/py-mdanalysis/package.py

```diff
@@ -18,141 +18,52 @@ class PyMdanalysis(PythonPackage):
-    depends_on("py-griddataformats@0.4.0:", when="@0.17.0:", type=("build", "run"))
+    depends_on("py-numpy@1.21.0:", when="@2.5.0:", type=("build", "run"))
+    depends_on("py-numpy@1.20.0:", type=("build", "run"))
 
-    depends_on("py-mmtf-python@1.0.0:", when="@0.16.0:", type=("build", "run"))
+    depends_on("py-biopython@1.80:", type=("build", "run"))
+    depends_on("py-networkx@2.0:", type=("build", "run"))
+    depends_on("py-griddataformats@0.4.0:", type=("build", "run"))
+    depends_on("py-mmtf-python@1.0.0:", type=("build", "run"))
+    depends_on("py-joblib@0.12:", type=("build", "run"))
 
-    depends_on("py-joblib", when="@0.16.0:0.20.1", type=("build", "run"))
-    depends_on("py-joblib@0.12:", when="@1.0.0:", type=("build", "run"))
+    depends_on("py-scipy@1.5.1:", when="@2.5.0:", type=("build", "run"))
+    depends_on("py-scipy@1.5.0:", type=("build", "run"))
 
-    depends_on("py-scipy", when="@:0.16.1+analysis", type=("build", "run"))
-    depends_on("py-scipy", when="@0.16.2:0.17.0", type=("build", "run"))
-    depends_on("py-scipy@1.0.0:", when="@0.18.0:1.0.0", type=("build", "run"))
-    depends_on("py-scipy@1.5.0:", when="@2.4.0:", type=("build", "run"))
-
-    depends_on("py-matplotlib", when="@:0.15.0+analysis", type=("build", "run"))
-    depends_on("py-matplotlib@1.5.1:", when="@0.16.0:0.16.1+analysis", type=("build", "run"))
-    depends_on("py-matplotlib@1.5.1:", when="@0.16.2:", type=("build", "run"))
-
-    depends_on("py-tqdm@4.43.0:", when="@1.0.0:", type=("build", "run"))
-
-    depends_on("py-threadpoolctl", type=("build", "run"), when="@2.4.0:")  # Requirement from
-
-    depends_on("py-packaging", when="@2.4.2:", type=("build", "run"))
-
-    depends_on("py-fasteners", when="@2.4.2:", type=("build", "run"))
-
-    depends_on("py-gsd@1.4.0:", when="@0.17.0:1.0.0", type=("build", "run"))
-    depends_on("py-gsd@1.9.3:", when="@2.4.0:", type=("build", "run"))
+    depends_on("py-matplotlib@1.5.1:", type=("build", "run"))
+    depends_on("py-tqdm@4.43.0:", type=("build", "run"))
+    depends_on("py-threadpoolctl", type=("build", "run"))
+    depends_on("py-packaging", type=("build", "run"))
+    depends_on("py-fasteners", type=("build", "run"))
+    depends_on("py-gsd@1.9.3:", type=("build", "run"))
 
     # extra_format (extras_require)
-    depends_on("py-netcdf4@1.0:", when="+amber", type=("build", "run"))
     depends_on("py-netcdf4@1.0:", when="+extra_formats", type=("build", "run"))
-
-    depends_on("hdf5", when="+amber", type=("run"))
     depends_on("py-h5py@2.10:", when="+extra_formats", type=("build", "run"))
-
-    depends_on("py-pytng@0.2.3:", when="@2.4.0:+extra_formats", type=("build", "run"))
-
-    depends_on("py-chemfiles@0.10:", when="@2.4.0:+extra_formats", type=("build", "run"))
-
-    depends_on("py-pyedr@0.7.0:", when="@2.4.0:+extra_formats", type=("build", "run"))
+    depends_on("py-pytng@0.2.3:", when="+extra_formats", type=("build", "run"))
```

Comment:
I would move this to the end of the list because it appears last in pyproject.toml. Not required though, just makes it easier to review.